### PR TITLE
fix: image naming conflicts

### PR DESCRIPTION
**Pull request by mbssantos**
Targeting #41

### CODE REVIEW

File: site/builder/copyReadmeAssets.js [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
 const apiDistFolder = path.resolve(root, 'site/dist/api');
 
 const imageGlob = 'node_modules/d3fc-*/**/*.png';
 
-function ensureExists(pathname, cb) {
-  fs.mkdir(path.dirname(pathname), (err) => {
-    if (!err || err.code === 'EEXIST') {
-      return cb();
+function ensureExists(pathname, callback) {
+  mkdirp(path.dirname(pathname), (err) => {
```
Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: As the rest of this file uses promises already, I'd rather see non-promise code wrapped in a promise ASAP to allow easier composition and refactoring later. So in this case I'd expect to see `ensureExists` return a promise rather than accept a callback.

File: site/builder/copyReadmeAssets.js [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
+      throw err;
     }
-    throw err;
+    return callback();
   });
 }
 
 const getImageFilename = (pathname) => {
-  return path.join(apiDistFolder, pathname.replace(new RegExp('node_modules/[^`/]*/'), ''));
+  return path.join(apiDistFolder, pathname.replace(new RegExp('node_modules/'), ''));
```
Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: I have a preference for the regex literal syntax, it's more terse.

Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: Also if a fat-arrow amounts to `=> { return expression; }` we should simplify it to `=> expression`. Are these files being linted?

File: site/builder/copyReadmeAssets.js [hunk 3]
```diff
@@ -44,7 +43,8 @@ export default (data) =>
 
       return Promise
         .all(readPromises)
-        .then(() => resolve(data))
+        .then(() => { resolve(data) } )
```
Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: Is this change required? This whole function is very convoluted, can it be expressed more simply?

Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: (I'm aware your changes weren't responsible :smiley: )

File: site/builder/parseReadmes.js [hunk 4]
```diff
@@ -16,11 +16,24 @@ function parseBlock(level, content, title, options) {
   const children = [];
   const regex = createBlockPattern(level);
 
+  if(content && title && title.indexOf('d3fc') >= 0) {
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: I don't think the whitespace conforms to our lint rules, can you ensure these files are covered by the linting?

File: site/builder/parseReadmes.js [hunk 5]
```diff
@@ -16,11 +16,24 @@ function parseBlock(level, content, title, options) {
...
+    var fileNames = content.match(/\(([^)]+\.png)\)/g);
+    content = content.replace(new RegExp(`screenshots/`, 'g'), `${title}/screenshots/`)
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: regex literal

File: site/builder/parseReadmes.js [hunk 6]
```diff
@@ -16,11 +16,24 @@ function parseBlock(level, content, title, options) {
   const children = [];
   const regex = createBlockPattern(level);
 
+  if(content && title && title.indexOf('d3fc') >= 0) {
+    var fileNames = content.match(/\(([^)]+\.png)\)/g);
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: Use let/const

File: site/builder/parseReadmes.js [hunk 7]
```diff
@@ -16,11 +16,24 @@ function parseBlock(level, content, title, options) {
...
+    content = content.replace(new RegExp(`screenshots/`, 'g'), `${title}/screenshots/`)
+
+    if(fileNames) {
+      fileNames.forEach(function(name) {
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: Use fat-arrow

File: site/builder/parseReadmes.js [hunk 8]
```diff
@@ -16,11 +16,24 @@ function parseBlock(level, content, title, options) {
...
+        content = content.replace(name, `(${title}/${name.replace(/\(|\)/g, '')})`);
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: I think all of this can be replaced with a single call to `content.replace`, happy to work it through in person?

**Comment by ColinEberhardt** [comment]
@mbssantos - remember to add a comment to the PR when you want people to review it again :-)

**Comment by mbssantos** [comment]
@chrisprice PR is ready for another go!

File: site/builder/copyReadmeAssets.js [hunk 9]
```diff
@@ -1,30 +1,29 @@
...
 const imageGlob = 'node_modules/d3fc-*/**/*.png';
 
-function ensureExists(pathname, cb) {
-  fs.mkdir(path.dirname(pathname), (err) => {
-    if (!err || err.code === 'EEXIST') {
-      return cb();
-    }
-    throw err;
-  });
+function ensureExists(pathname, callback) {
```
Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: pedant...
```js
const ensureExists = new Promise((resolve) => {
  mkdirp(path.dirname(pathname), (err) => {
    if (err) {
      throw err;
    }
    resolve();
  });
});
```

File: site/builder/parseReadmes.js [hunk 10]
```diff
@@ -11,16 +11,22 @@ const createBlockPattern = (level) => new RegExp(
...
 
 function parseBlock(level, content, title, options) {
-  // console.log(level, content, title, options);
   let match, firstChildIndex;
   const children = [];
   const regex = createBlockPattern(level);
 
+  if (content && title && title.indexOf('d3fc') >= 0) {
+    content = content.replace(/"screenshots\//g, `"${title}/screenshots/`);
+    content = content.replace(/((\(.+)\.png)/g, (match, fileName) =>  `(${title}/${fileName.replace(/\(|\)/g, '')})`);
```
Comment (chrisprice) [inline site/builder/parseReadmes.js]: Nice

File: site/builder/copyReadmeAssets.js [hunk 11]
```diff
@@ -1,30 +1,24 @@
...
-    if (!err || err.code === 'EEXIST') {
-      return cb();
-    }
-    throw err;
-  });
-}
+const ensureExists = (pathname, callback) =>
+  new Promise((resolve, reject) =>
+    mkdirp(path.dirname(pathname), (err) =>
+      err ? reject(err) : resolve()
```
Comment (chrisprice) [inline site/builder/copyReadmeAssets.js]: Nice
